### PR TITLE
Listing block template

### PR DESCRIPTION
**Pull request by giuliaghisini**
By default there's only one template for listing-block, that is overridable from @package/customizations/components/Blocks/manage/Listing/DefaultTemplate.jsx. 

If you want to add other templates, you could add them to config.js in listing block configuration in templates field. Here an example:
```

import TempalteOneListingBlockTemplate from '@package/components/.........../Templates/TemplateOne/TempalteOneListingBlockTemplate';
import TempalteTwoListingBlockTemplate from '@package/components/.........../Templates/TemplateOne/TempalteTwoListingBlockTemplate';

const customBlocks = {
.......
listing: {
    ...defaultBlocks.blocksConfig.listing,
    templates: {
      ...defaultBlocks.blocksConfig.listing.templates,
      templateOne: { label: 'Template 1', template: TemplateOneListingBlockTemplate },
      templateTwo: { label: 'Template 2', template: TemplateTwoListingBlockTemplate },
    },
  }
};
export const blocks = {
  ...defaultBlocks,
  blocksConfig: { ...defaultBlocks.blocksConfig, ...customBlocks },
};
```

Then you have to create your own Template components under your project.

Template selector appears in sidebar, only if there's more of one template specified in config.

### CODE REVIEW

File: src/components/manage/Blocks/Listing/ListingStyle.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { blocks } from '~/config';
+import { compose } from 'redux';
+import withObjectBrowser from '@plone/volto/components/manage/Sidebar/ObjectBrowser';
+import TemplateWidget from './TemplateWidget';
+
+const ListingStyle = ({ data, block, onChangeBlock, required = false }) => {
+  const templatesConfig = blocks?.blocksConfig?.listing?.templates;
```
Comment (pnicolli) [inline src/components/manage/Blocks/Listing/ListingStyle.jsx]: Same as before, if we remove the ``?.`` there, we should remove it here as well.

File: src/components/manage/Blocks/Listing/ListingStyle.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+};
+
+ListingStyle.propTypes = {
+  data: PropTypes.objectOf(PropTypes.any).isRequired,
+  block: PropTypes.string.isRequired,
+  onChangeBlock: PropTypes.func.isRequired,
+  openObjectBrowser: PropTypes.func.isRequired,
+};
+
+export default compose(withObjectBrowser)(ListingStyle);
```
Comment (pnicolli) [inline src/components/manage/Blocks/Listing/ListingStyle.jsx]: Is the object browser possibly here by mistake? Or is it used somewhere I didn't notice?

File: src/components/manage/Blocks/Listing/TemplateWidget.jsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+};
+
+TemplateWidget.propTypes = {
+  data: PropTypes.objectOf(PropTypes.any).isRequired,
+  block: PropTypes.string.isRequired,
+  onChangeBlock: PropTypes.func.isRequired,
+  openObjectBrowser: PropTypes.func.isRequired,
+};
+
+export default compose(withObjectBrowser, injectIntl)(TemplateWidget);
```
Comment (pnicolli) [inline src/components/manage/Blocks/Listing/TemplateWidget.jsx]: Same question as before, maybe I am missing why the object browser is needed, or is it just a copy-paste leftover?

File: src/components/manage/Blocks/Listing/ListingStyle.jsx (hunk 2 above)
Comment (giuliaghisini) [reply src/components/manage/Blocks/Listing/ListingStyle.jsx]: no it's a mistake 😢

**Comment by sneridagh** [comment]
@pnicolli needs fix the i18n, can you do it? then we will merge

**Comment by giuliaghisini** [comment]
> @pnicolli needs fix the i18n, can you do it? then we will merge

Hi Victor, I merged i18n files but checks still fails because no new line is present at the end of locales *lang*.json files.. maybe the yarn run prettier command or other command in checks changes these files and checks fails..

**Comment by tisto** [comment]
@giuliaghisini i18n is fixed. snapshot tests are failing now. pls check.

**Comment by sneridagh** [comment]
Sync'd with master. I'll merge after green. 

@giuliaghisini please add your name in CHANGELOG too. I did it for you in this one :)

**Comment by giuliaghisini** [comment]
> Sync'd with master. I'll merge after green.
> 
> @giuliaghisini please add your name in CHANGELOG too. I did it for you in this one :)

ok! next times i'll do it! thanks so much for this one!

**Pull request merged**